extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntity.java

```diff
@@ -39,5 +41,25 @@ public String toString() {
         return this.getClass().getSimpleName() + "<" + id + ">";
     }
 
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (Objects.isNull(object) || getClass() != object.getClass()) {
```

Comment:
use `object == null`.

Reply:
I'm so sorry, but why? Objects.isNull(object) executes it:
public static boolean isNull(Object obj) { return obj == null; }

Objects.isNull(object) is more semantic.

Reply:
https://github.com/quarkusio/quarkus/pull/8241#discussion_r399723086